### PR TITLE
Append leftovers of FromBase64Transform's input buffer to the transform buffer

### LINKED ISSUE #76918: IndexOutOfRangeException on decoding base64 bytes with LF characters included (>= .NET 5 only)
### Description

I have upgraded runtime from netcoreapp3.1 to NET 6 and the code which was transforming base64 bytes containing LF characters to string without errors started to crash with IndexOutOfRangeException. Behavior is the same for Windows and Linux platforms.

### Reproduction Steps

```
[TestFixture]
public class Fixture
{
    [Test]
    public void Crashes_on_runtime_greater_or_equal_to_NET_5()
    {
        var txt = "YWJj\nZGVm"; // abc\ndef
        var base64Bytes = Encoding.UTF8.GetBytes(txt);
        var stream = new MemoryStream(base64Bytes);
        var base64Transform = new FromBase64Transform();
        var cryptoStream = new CryptoStream(stream, base64Transform, CryptoStreamMode.Read);

        var result = new MemoryStream();
        cryptoStream.CopyTo(result);

        Console.WriteLine(Encoding.UTF8.GetString(result.ToArray()));
    }
}
```

### Expected behavior

"abcdef" in the output, which is happening for example on netcoreapp3.1 runtime.

### Actual behavior

```
System.IndexOutOfRangeException : Index was outside the bounds of the array.
   at System.Security.Cryptography.FromBase64Transform.TransformFinalBlock(Byte[] inputBuffer, Int32 inputOffset, Int32 inputCount)
   at System.Security.Cryptography.CryptoStream.ReadAsyncCore(Memory`1 buffer, CancellationToken cancellationToken, Boolean useAsync)
   at System.Security.Cryptography.CryptoStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.Security.Cryptography.CryptoStream.CopyTo(Stream destination, Int32 bufferSize)
   at System.IO.Stream.CopyTo(Stream destination)
   at ClassLibrary1.Fixture.Crashes_on_runtime_greater_or_equal_to_NET_5() in E:\cm_1\drive\ClassLibrary1\Class1.cs:line 20
```

### Regression?

Regression happened after the upgrade from netcoreapp3.1 to NET 5/NET 6.

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by jozkee**
Fixes https://github.com/dotnet/runtime/issues/76918,

As @gfoidl pointed out on https://github.com/dotnet/runtime/issues/76918#issuecomment-1276070344; we need to append _inputBuffer followed by the CryptoStream's inputBuffer to the transform buffer in order to properly decode. Currently we were only appending the latter and on cases where there was only 1 byte we were causing a `IndexOutOfRangeException` on `GetOutputSize`.

Could there be more severe bugs related to this issue that require additional test cases?

### CODE REVIEW

**Comment by vcsjones** [comment]
@Jozkee these changes appear to have introduced a regression. I did some light fuzzing of these changes and distilled a repro down to:

```C#
  [Fact]
  public void fuzzcase()
  {
      ICryptoTransform transform = new FromBase64Transform();
      string base64 = "s3 DhwGM/aBkdcMBqsXMLINDLvPH9htGe01DfPYYHB4P4mSdH42fDSo9kX5Z89hmBT8IlrDjvAkHcYMxTfwW3l8XUmKVossWvznTtByq5bIQwkc9WJu9om49Irk7PvATqbiHmcIH+KpPLWL1UBtXG3JKfWydpVwjGN/a2F+zUEYM=";
      byte[] input = System.Text.Encoding.ASCII.GetBytes(base64);
      byte[] destination = new byte[128];
      transform.TransformBlock(input, 0, 1, destination, 0);
      transform.TransformBlock(input, 1, 1, destination, 1);
      transform.TransformBlock(input, 2, 1, destination, 2);
      transform.TransformBlock(input, 3, 1, destination, 3);
      // etc
  }
```

In `main`, this does not throw an exception. In the PR, this will fall with:

<pre>
Error Message:
   System.ArgumentException : Destination is too short. (Parameter 'destination')
  Stack Trace:
     at System.Security.Cryptography.FromBase64Transform.TransformBlock(Byte[] inputBuffer, Int32 inputOffset, Int32 inputCount, Byte[] outputBuffer, Int32 outputOffset) in /Users/vcsjones/Projects/runtime/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Base64Transforms.cs:line 156
   at System.Security.Cryptography.Tests.Base64TransformsTests.fuzzcase() in /Users/vcsjones/Projects/runtime/src/libraries/System.Security.Cryptography/tests/Base64TransformsTests.cs:line 128
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr) in /Users/vcsjones/Projects/runtime/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInvoker.cs:line 59
</pre>

We should fix and add some additional test cases since an existing test did not catch this.

**Comment by Jozkee** [comment]
@vcsjones the problem was when inputBuffer was small enough to save it in the ICryptoTransform buffer in more than one call. Since now we append both buffers before "checking if bytesToTransform < InputBlockSize else save them", I just removed the `AsSpan` that was acting as a `Slice` that was causing the problem. 

I also added a test for `ReadOneByteAtTheTime` with multiple sizes of inputBuffer.

**Comment by vcsjones** [comment]
I haven't been able to break this with the latest changes, so, LGTM.

File: src/libraries/System.Security.Cryptography/tests/Base64TransformsTests.cs [hunk 1]
```diff
@@ -361,5 +367,27 @@ private static int ReadAll(Stream stream, Span<byte> buffer)
 
             return totalRead;
         }
+
+        [Theory, MemberData(nameof(TestData_All_Ascii))]
+        public void ReadOneByteAtTheTime(string expected, string data)
```
Comment (wfurt) [inline src/libraries/System.Security.Cryptography/tests/Base64TransformsTests.cs:372]: I'm wondering if we should also feed it byte by byte to be sure the incomplete input is handled properly.

**Review by wfurt — APPROVED** [review]
LGTM.
We should probably rebase it or sync up as it seems pretty old and there some some strange infrastructure failures.

File: src/libraries/System.Security.Cryptography/tests/Base64TransformsTests.cs (hunk 1 above)
Comment (Jozkee) [reply src/libraries/System.Security.Cryptography/tests/Base64TransformsTests.cs:372]: It is also feeding one byte at a time. On Line 384 `inputCount` argument is 1:
```
written += transform.TransformBlock(inputBytes, inputBytes.Length - remaining, 1, outputBytes, written);
```

**Pull request merged**